hera_mc/cm_handling.py

```diff
@@ -160,13 +140,11 @@ def find_ports(self, connection_dossier):
 
         input_ports = []
         output_ports = []
-        for k, v in connection_dossier.iteritems():
-            if k in self.non_class_connection_dossier_entries:
-                continue
-            if ':up:' in k and v.downstream_input_port not in input_ports:
-                input_ports.append(v.downstream_input_port)
-            elif ':down:' in k and v.upstream_output_port not in output_ports:
-                output_ports.append(v.upstream_output_port)
+        for k, c in connection_dossier['connections'].iteritems():
+            if c.downstream_input_port not in input_ports:
+                input_ports.append(c.downstream_input_port)
+            elif c.upstream_output_port not in output_ports:
+                output_ports.append(c.upstream_output_port)
```

Comment:
Err, I think the elif is doing more work than that. Because it's the `else` branch, the loop will only register the output ports of parts that it *hasn't* registered the input ports for. The logic is equivalent to:

```
if downstream_input not in input_ports:
   input_ports.append(downstream_input)
else:
   if upstream_output not in output_ports:
         output_ports.append(upstream_output)
```

Is that really intended, or should it be the following?

```
if downstream_input not in input_ports:
   input_ports.append(downstream_input)
if upstream_output not in output_ports:
   output_ports.append(upstream_output)
```